crates/ra_tt/src/lib.rs

```diff
@@ -54,23 +54,49 @@ pub struct Subtree {
     pub token_trees: Vec<TokenTree>,
 }
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, Debug, Hash)]
 pub enum Delimiter {
-    Parenthesis,
-    Brace,
-    Bracket,
+    Parenthesis(TokenId),
+    Brace(TokenId),
+    Bracket(TokenId),
     None,
 }
 
+impl Delimiter {
+    pub fn id(&self) -> Option<TokenId> {
+        match self {
+            Delimiter::Parenthesis(id) => Some(*id),
+            Delimiter::Brace(id) => Some(*id),
+            Delimiter::Bracket(id) => Some(*id),
+            Delimiter::None => None,
+        }
+    }
+}
+
+impl PartialEq for Delimiter {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Delimiter::Parenthesis(..), Delimiter::Parenthesis(..))
+            | (Delimiter::Brace(..), Delimiter::Brace(..))
+            | (Delimiter::Bracket(..), Delimiter::Bracket(..)) => true,
```

Comment:
ignoring the token ID here seems weird, do we do that for the other token types?

Reply:
> ignoring the token ID here seems weird, do we do that for the other token types?

No, you are right, but this is the fastest method to let all test passes :) (There is a lot of code inside macro rules matching which compare 2 delimiters). I should handle it properly.

Reply:
Fixed, make it a explict method instead.

Reply:
I think that maybe https://github.com/rust-analyzer/rust-analyzer/pull/2545/files#r357562329 helps with this? `delim1.kind == delim2.kind` should read better.